### ISSUE TITLE
Evaluate and Update or Remove apparently obsolete services

### ISSUE THREAD

**Issue by Asher-** [adium/adium#18]
AIM, possibly others. 

1. determine which are obsolete
2. remove from code
3. ensure code still works

[Edited by @RobFog]
List of services to consider removing:
* AIM (see also #20)
* MSN
* GoogleTalk (?) (see also #1)
* Yahoo! and Yahoo! Japen
* Facebook
* LiveJournal
* MobileMe

**Comment by stephenchaffins** [comment]
Please dont remove aim. I actually just made a feature request for a modification to it to make it useful (before seeing this issue).

**Comment by RobFog** [comment]
I am skeptical that there are good reasons to keep AIM for AIM Phoenix. Let's discuss AIM at #20, the issue you opened.

**Comment by RobFog** [comment]
Other services to remove would be:
* MSN
* GoogleTalk (?)
* Yahoo! and Yahoo! Japen
* Facebook
* LiveJournal
* MobileMe

**Comment by estella144** [comment]
_Facebook_?

**Comment by Asher-** [comment]
Facebook cut off access to their messenger API quite a while back because it was undermining exposure to ads. 

At some point, I noticed it happened to be working again. I don't think it was a policy reversal on their part, so I'm not sure what the resulting state was. 

Anyone know?

**Comment by RobFog** [comment]
I'm sure Adium's current Facebook implementation doesn't work. This libpurple plugin does and is being actively developed: https://github.com/dequis/purple-facebook works. We should consider removing built-in support and pointing to the plugin instead.

**Comment by estella144** [comment]
@RobFog Ok, then how is MSN obsolete?

**Comment by RobFog** [comment]
I'm sorry but I don't understand the question. MSN has been dead for a long time: https://en.wikipedia.org/wiki/Windows_Live_Messenger

**Comment by jas8522** [comment]
A bunch of this is done in my fork here: https://github.com/jas8522/adium

But I'm stuck where the main project is stuck: I can't tell the results of my changes without being able to compile it.

**Comment by estella144** [comment]
@RobFog you are correct. In the Discontinuation section of that article, it says:

> On the same day [February 15 2013], Microsoft announced its plans to phase out Messenger: The process would start on April 8, 2013, with English users and would end on April 30, 2013.

It needs to be dropped.

**Comment by estella144** [comment]
Also, at [https://en.wikipedia.org/wiki/Yahoo!_Messenger#Adoption](url) it says that:

> At the time of Yahoo! Messenger's closure in 2018, it remained popular in Vietnam.

What are the implications of that?

**Comment by jas8522** [comment]
> I'm sure Adium's current Facebook implementation doesn't work. This libpurple plugin does and is being actively developed: https://github.com/dequis/purple-facebook works. We should consider removing built-in support and pointing to the plugin instead.

Shall we create a separate issue to track that move to use FB messenger with libpurple?

**Comment by RobFog** [comment]
I think we need to have a general discussion about which services to support etc. It might be better to leave that discussion for until we have released the minor update with build fixes and the scrolling bug fix.

> https://en.wikipedia.org/wiki/Yahoo!_Messenger#Adoption



> Also, at [https://en.wikipedia.org/wiki/Yahoo!_Messenger#Adoption](url) it says that:
> 
> > At the time of Yahoo! Messenger's closure in 2018, it remained popular in Vietnam.
> 
> What are the implications of that?

None. It's dead.

**Comment by Neustradamus** [comment]
I think that Google Talk always works, no?
The only point is that Google Talk has disabled the XMPP Federation when:

May 19, 2014 - permanent upgrade to encrypted network, coinciding with Open Discussion Day:
- http://opendiscussionday.org/

**Comment by estella144** [comment]
I don't understand @Neustradamus . By the way @RobFog , what do you mean by the "scrolling bug"?

**Comment by estella144** [comment]
> > I'm sure Adium's current Facebook implementation doesn't work. This libpurple plugin does and is being actively developed: https://github.com/dequis/purple-facebook works. We should consider removing built-in support and pointing to the plugin instead.
> 
> Shall we create a separate issue to track that move to use FB messenger with libpurple?

Probably @jas8522 .

**Comment by estella144** [comment]
At the end of the UTC day (literally), we all agree these services should be removed:

- Facebook
- MSN
- Yahoo! Messenger

Any others can be discussed tomorrow.

**Comment by Neustradamus** [comment]
MSN/WLM -> Escargot : https://escargot.log1p.xyz/ + https://escargot.log1p.xyz/news + https://gitlab.com/escargot-chat
AIM/Yahoo/ICQ -> http://iwarg.ddns.net/phoenix/ + http://iwarg.ddns.net/phoenix/index.php?action=faqpage + https://www.phoenix.wildman-productions.org/

**Comment by Asher-** [comment]
Neustradamus - could you assemble details on current working state / updates required to work on either of those platforms?

**Comment by estella144** [comment]
As @RobFog said:

> List of services to consider removing:
> 
> * AIM (see also #20)
> * MSN
> * GoogleTalk (?) (see also #1)
> * Yahoo! [Messenger] and Yahoo! Japan
> * Facebook
> * LiveJournal
> * MobileMe

We haven't discussed LiveJournal and MobileMe yet. We have discussed:

* AIM (see also #20)
* MSN
* GoogleTalk (?) (see also #1)
* Yahoo! [Messenger] and Yahoo! Japan
* Facebook

The general opinion is that these should be removed. @Asher- can you evaluate the status of the other ones?

**Comment by ChrisForsythe** [comment]
@Asher- is busy with other things. I would suggest someone else handle reviewing each protocol for removal or not. Once a full list of the protocols with justification for each modification or keeping them is taken care of, it'll be easier for him or someone else to make a determination on the how if everyone on the agrees on the what.

So if something like this were filled out by someone it would be helpful:

| Service | Officially Supported | prpl status | Proposed Action | Reason |
| --- | --- | --- | --- | --- |
| AIM | No | something | Hold | No official support, there are some unofficial servers which require investigation |
| MSN | No | something else | Remove | Some reason |
| IRC | Yes | prpl is currently being updated | Keep | Something else |

This table is just an example. We can comment back and forth on each protocol individually if this doesn't make sense, but it would be more organized to get a summary table together showing the whole picture in one easy to use table.


@estella144 and @Neustradamus if you would like this might be up your alley but it's just a thought.

**Comment by estella144** [comment]
@ChrisForsythe I am quite interested, but I think this should be done by **2** people. @Neustradamus are *you* interested?

**Comment by estella144** [comment]
| Service        | Officially Supported  | prpl status | Proposed action | Reason |
| ------------- | ------------- | ----- | :----------: | ---- |
| AIM      | No | [To be filled in] | Remove | Dead for a long time |
| MSN      | No | [To be filled in] | Remove | Dead for a long time |
| IRC | Yes  | [To be filled in] | Keep | Still active (many servers, we use it) |
| Facebook | No | API disabled | Remove | API has been disabled by Facebook |
| XMPP | Yes | [To be filled in] | Keep | |

**Comment by estella144** [comment]
@ChrisForsythe which repo is prpl?

**Comment by Asher-** [comment]
https://keep.imfreedom.org/pidgin/pidgin/ - you'll have to use hg.

**Comment by estella144** [comment]
@Asher- hg? I can't install hg.

**Comment by ChrisForsythe** [comment]
> @Asher- hg? I can't install hg.

hg is mercurial. Really you just need to know what a prpl is and then how to check the status. A prpl is a pidgin protocol library. So aim would have one, irc, etc etc.

Finding out the status might be as simple as loading up pidgin and seeing what it offers, can connect, etc. rather than going through the mercurial repository. It could be handing that off to someone else while checking for official service statuses on each available protocol.

**Comment by estella144** [comment]
@ChrisForsythe I can't seem to get them, so can you work on it?

**Comment by ChrisForsythe** [comment]
> @ChrisForsythe I can't seem to get them, so can you work on it?

Please come to #adium-devl on freenode.

**Comment by estella144** [comment]
Thanks.

**Comment by estella144** [comment]
@ChrisForsythe is busy.

**Comment by Neustradamus** [comment]
About AIM/Yahoo/ICQ and MSN, I do not use but there are supported servers...
Maybe good to keep, maybe a little change is needed...
It will be bad to remove it...
- https://github.com/adium/adium/issues/18#issuecomment-795291127

I will try to contact users...

**Comment by Neustradamus** [comment]
I do not use IRC, I use XMPP, example Pidgin MUC Room is:
xmpp:devel@conference.pidgin.im?join

Recently, Pidgin has launched a PidginChat: https://pidgin.im/about/pidginchat/
It is XMPP service without the XMPP Federation.

**Comment by toeserve** [comment]
We definitely should not remove support for MSN, AIM, or Yahoo, as new servers and services have been developed which support these protocols, so Adium users need support for these protocols to connect to these services. 

Escargot supports Yahoo and MSN:  

https://escargot.log1p.xyz/

AIM Phoenix supports AIM:
http://iwarg.ddns.net/phoenix/index.php

That these new implementations of these protocols exist means Adium users can benefit from the continued support for these protocols to connect to these servers.

Also Google Talk is working fine. All Google Talk is, is an XMPP service anyway.

**Comment by jas8522** [comment]
@toeserve seems reasonable to me, particularly for AIM / Phoenix.

Generally: wasn't Yahoo/MSN already removed a while ago? I don't see it in the list Adium presents to me when trying to add a new account.

- MobileMe: I can't see any reason why this should still exist
- Google Talk: may as well remain while it continues to work (I kinda thought it would be dead by now). Perhaps it should be rebranded to Google Hangouts or Google Hangouts Chat? It would make sense given that Wikipedia says the following which implies it only works with Google Hangouts: "The Google Talk App for Android and the Google Chat tool in Gmail were discontinued on June 26, 2017 and no longer function. Users may still continue to use third-party XMPP clients to connect to the legacy Google talk server, but only for 1-on-1 chat with Hangouts users."
- Facebook: As this issue is to track removal of services, I still think we should create a new issue to track the replacement of the existing implementation (which people have indicated above does not work) with the working [purple-facebook library here](https://github.com/dequis/purple-facebook), as @RobFog indicated above.

**Comment by estella144** [comment]
@Neustradamus we should not be supporting ICQ anyway. As stated in https://en.wikipedia.org/wiki/ICQ#Clients :

> AOL's OSCAR network protocol used by ICQ is proprietary and using a third-party client is a violation of [the] ICQ Terms of Service. Nevertheless, a number of third-party clients have been created by using reverse-engineering and protocol descriptions.

We don't want to get in trouble, so drop that.

**Comment by Asher-** [comment]
Wikipedia aside, I don't see any evidence of that in ICQ's actual terms of service. Maybe at one point this was true.

**Comment by estella144** [comment]
@Asher- let me check.

**Comment by estella144** [comment]
@Asher- the terms are at https://help.mail.ru/legal/terms/icq/eu/ircterms - it says:

> YOU WILL NOT: ... (7) translate, decompile, reverse engineer, disassemble, modify, copy, alter, merge into other software, reproduce, rent, lease, lend, distribute, remarket or otherwise dispose of The IrCQ-Net or any part thereof.

That is the evidence.

**Comment by Asher-** [comment]
That's totally unenforceable and has no relation to Adium.

Decompile, disassemble, modify, copy, alter, merge, rent, lease, lend, distribute, remarket => nothing to do with Adium.

Reverse engineer => requires precise definition, most of which (particularly in EU) are explicitly protected actions (reverse engineering for compatibility is specifically protected by EU law). 

Reproduce => arguably, except that would require that the open source code actually corresponded to proprietary code (thus _reproducing_ it). So long as the open source code is not based on the proprietary code, it is not a reproduction.

**Comment by Neustradamus** [comment]
My remark about ICQ is linked to:
- http://iwarg.ddns.net/phoenix/

"Rudimentary Support for ICQ clients in the 2000b / 6.5 range is now enabled. ICQ support is still in development,
so ICQ clients will still be in a rather crippled state."

Note: AIM and ICQ, it is OSCAR;
- https://en.wikipedia.org/wiki/OSCAR_protocol
- https://en.wikipedia.org/wiki/TOC_protocol

**Comment by estella144** [comment]
@Asher- we will need to use the protocol (which is a part of IrCQ), so that falls under disassembling and copying (and arguably reproducing). The terms are enforceable.

@Neustradamus you said, "... ICQ clients will still be in a rather crippled state." Even more of a reason to not do that.

**Comment by Asher-** [comment]
That's not correct.

**Comment by estella144** [comment]
@Asher- in what way?

**Comment by Asher-** [comment]
To answer, we would have to be talking about a specify body of law. In general, using a library—even if it has been created by way of reverse engineering, decompiling, etc—is not equivalent to us reverse engineering, decompiling, etc. There are questions at stake, such as whether the result "reproduces" the code in question, but the bottom line is that none of us have done any of the things that would be violations of the terms of service. Even if someone else violated the terms of service to create the code in question, if _using_ the code in question does not _directly_ violate the terms of service, there is zero issue. If it does violate the terms of service, that gets into a whole range of other questions, including: did the individual ostensibly violating the terms of service ever agree to the terms of service?

In short, you are making legal assumptions where you don't have a basis to make them, and you are therefore making incorrect conclusions about necessary outcomes.

**Comment by estella144** [comment]
This is getting off-topic @Asher-. I *will* update the table, but I have yet to receive anything about prpl status.

**Comment by Asher-** [comment]
The best way to handle this would be to document your concerns about potential terms of service / legal issues with 1. quotation from corresponding sources (as you've provided here) 2. your interpretation of how to understand how that impacts us 3. possible ambiguities or alternative understandings. 

The law is not a fixed thing. In no case does it simply exist. It is derived from contracts, which are written documents that have to be interpreted. There are always multiple possible interpretations, so it is important to examine each before assuming one to be the relevant case.

**Comment by estella144** [comment]
I will. Have any ambiguities or alternative understandings been identified @Asher- ?

**Comment by thomasrussellmurphy** [comment]
As a 10+ year user of the GoogleTalk account type, please ensure that any required migration is clear and usable.

**Comment by Animadoria** [comment]
MSN is definitely not obsolete -- [Escargot](https://escargot.chat) fully supports up to protocol 18 (MSNP18).

Libpurple seems to have removed MSN support. A Pidgin plugin for MSN that I've patched is here: https://github.com/Animadoria/msn-pecan - maybe it'd be helpful. Would love to help, but my Xcode / Objective-C knowledge is non existant, and I don't think I can even compile Adium on my M1 Mac.
